osbs/utils.py

```diff
@@ -190,72 +192,120 @@ def buildconfig_update(orig, new, remove_nonexistent_keys=False):
                 orig[k] = v
 
 
-def subprocess_check_output_with_retry(args, max_retries=0, backoff_factor=0):
+@contextlib.contextmanager
+def checkout_git_repo(git_url, target_dir=None, commit=None, retry_times=GIT_MAX_RETRIES,
+                      branch=None, depth=None):
+    """
+    clone provided git repo to target_dir, optionally checkout provided commit
+    yield the ClonedRepoData and delete the repo when finished
+
+    :param git_url: str, git repo to clone
+    :param target_dir: str, filesystem path where the repo should be cloned
+    :param commit: str, commit to checkout, SHA-1 or ref
+    :param retry_times: int, number of retries for git clone
+    :param branch: str, optional branch of the commit, required if depth is provided
+    :param depth: int, optional expected depth
+    :return: str, int, commit ID of HEAD
     """
-    Call subprocess.check_output with args, retrying max_retries times.
+    tmpdir = tempfile.mkdtemp()
+    target_dir = target_dir or os.path.join(tmpdir, "repo")
+    try:
+        yield clone_git_repo(git_url, target_dir, commit, retry_times, branch, depth)
+    finally:
+        shutil.rmtree(tmpdir)
+
 
-    :param args: list, arguments to be passed to subprocess.check_output
-    :param max_retries: int, maximum number of retries
-    :param backoff_factor: int, exponential backoff factor in seconds
-    :return: command output as byte string
+def clone_git_repo(git_url, target_dir=None, commit=None, retry_times=GIT_MAX_RETRIES, branch=None,
+                   depth=None):
+    """
+    clone provided git repo to target_dir, optionally checkout provided commit
+
+    :param git_url: str, git repo to clone
+    :param target_dir: str, filesystem path where the repo should be cloned
+    :param commit: str, commit to checkout, SHA-1 or ref
+    :param retry_times: int, number of retries for git clone
+    :param branch: str, optional branch of the commit, required if depth is provided
+    :param depth: int, optional expected depth
+    :return: str, int, commit ID of HEAD
     """
-    for counter in range(max_retries + 1):
+    retry_delay = GIT_BACKOFF_FACTOR
+    target_dir = target_dir or os.path.join(tempfile.mkdtemp(), "repo")
+    commit = commit or "master"
+    logger.info("cloning git repo '%s'", git_url)
+    logger.debug("url = '%s', dir = '%s', commit = '%s'",
+                 git_url, target_dir, commit)
+
+    cmd = ["git", "clone"]
+    if branch:
+        cmd += ["-b", branch, "--single-branch"]
+        if depth:
+            cmd += ["--depth", '{0}'.format(depth)]
```

Comment:
Any reason not to use `str(depth)` here?

Reply:
I'm bad at basic Python idiom.